File: 0007-reverse-integer/0007-reverse-integer.cs
public class Solution {
    public int Reverse(int x) {
        int reversed = 0;
        while(x != 0){
            int lastDigit = x % 10;
            if (reversed > int.MaxValue/10 || (reversed == int.MaxValue/10 && lastDigit > 7)){
                return 0;
            }
            if (reversed < int.MinValue/10 || (reversed == int.MinValue/10 && lastDigit < -8)){
                return 0;
            }
            reversed =  reversed*10 + lastDigit; 
            x /= 10;
        }
        return reversed;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        is_negative = False
        if x < 0:
            is_negative = True
        reversed_x = int(str(abs(x))[::-1])
        if reversed_x > 2**31 - 1:
            return 0
        return -reversed_x if is_negative else reversed_x